[Cocoa][MSE] Fix SourceBufferParserWebM logging

youssefsoli opened the pull request:
#### d6399ef829dd61ecf3b9f3058437e65ff78067b1
<pre>
[Cocoa][MSE] Fix SourceBufferParserWebM logging
<a href="https://bugs.webkit.org/show_bug.cgi?id=242720">https://bugs.webkit.org/show_bug.cgi?id=242720</a>

Reviewed by Jean-Yves Avenard.

Currently, the logger for SourceBufferParsers are not being set within
MediaSourcePrivateAVFObjC. Futhermore, the setLogger function within
SourceBufferParserWebM never sets its own logger and log identifier, leading
to no logs ever being outputted.

* Source/WebCore/platform/audio/cocoa/AudioFileReaderCocoa.cpp:
(WebCore::AudioFileReader::demuxWebMData const):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaSourcePrivateAVFObjC.mm:
(WebCore::MediaSourcePrivateAVFObjC::addSourceBuffer):
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.mm:
(WebCore::SourceBufferParserAVFObjC::~SourceBufferParserAVFObjC):
(WebCore::SourceBufferParserAVFObjC::appendData):
(WebCore::SourceBufferParserAVFObjC::flushPendingMediaData):
(WebCore::SourceBufferParserAVFObjC::setShouldProvideMediaDataForTrackID):
(WebCore::SourceBufferParserAVFObjC::shouldProvideMediadataForTrackID):
(WebCore::SourceBufferParserAVFObjC::resetParserState):
(WebCore::SourceBufferParserAVFObjC::invalidate):
(WebCore::SourceBufferParserAVFObjC::setLogger):
(WebCore::SourceBufferParserAVFObjC::didParseStreamDataAsAsset):
(WebCore::SourceBufferParserAVFObjC::didFailToParseStreamDataWithError):
(WebCore::SourceBufferParserAVFObjC::didProvideMediaDataForTrackID):
(WebCore::SourceBufferParserAVFObjC::willProvideContentKeyRequestInitializationDataForTrackID):
(WebCore::SourceBufferParserAVFObjC::didProvideContentKeyRequestInitializationDataForTrackID):
(WebCore::SourceBufferParserAVFObjC::didProvideContentKeyRequestSpecifierForTrackID):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::WebMParser::~WebMParser):
(WebCore::WebMParser::reset):
(WebCore::WebMParser::setLogger):
(WebCore::WebMParser::logChannel const):
(WebCore::WebMParser::invalidate):
(WebCore::SourceBufferParserWebM::~SourceBufferParserWebM):
(WebCore::SourceBufferParserWebM::setLogger):
(WebCore::SourceBufferParserWebM::setMinimumAudioSampleDuration):
(WebCore::SourceBufferParserWebM::logChannel const):
(WebCore::logChannel): Deleted.
(WebCore::logClassName): Deleted.
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.h:
(WebCore::WebMParser::TrackData::logChannel const):
(WebCore::WebMParser::VideoTrackData::logClassName const):
(WebCore::WebMParser::AudioTrackData::logClassName const):
(WebCore::WebMParser::VideoTrackData::logChannel const): Deleted.
(WebCore::WebMParser::AudioTrackData::logChannel const): Deleted.
(WebCore::WebMParser::loggerPtr const):
(WebCore::SourceBufferParserWebM::loggerPtr const):
(WebCore::WebMParser::logIdentifier const): Deleted.
(WebCore::SourceBufferParserWebM::logIdentifier const): Deleted.
* Source/WebKit/Shared/mac/MediaFormatReader/MediaFormatReader.cpp:
(WebKit::MediaFormatReader::nextLogIdentifier):
(WebKit::MediaFormatReader::logChannel):
(WebKit::nextLogIdentifier): Deleted.
(WebKit::logChannel): Deleted.
(WebKit::logClassName): Deleted.
* Source/WebKit/Shared/mac/MediaFormatReader/MediaFormatReader.h:

Canonical link: <a href="https://commits.webkit.org/252482@main">https://commits.webkit.org/252482@main</a>
</pre>

jyavenard commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.mm:
```diff
@@ -232,12 +231,14 @@ AtomString codec() const override
 
 SourceBufferParserAVFObjC::~SourceBufferParserAVFObjC()
 {
+    ALWAYS_LOG_IF_POSSIBLE(LOGIDENTIFIER);
```
why log only the destructor?

youssefsoli replied:
We don't have access to a logger in the constructor, it will never log, so instead we log in setLogger

jyavenard commented on Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.mm:
```diff
@@ -250,43 +251,50 @@ AtomString codec() const override
 
 #if !RELEASE_LOG_DISABLED
-void SourceBufferParserAVFObjC::setLogger(const Logger& logger, const void* logIdentifier)
+void SourceBufferParserAVFObjC::setLogger(const Logger& newLogger, const void* newLogIdentifier)
```
don't think we need to rename those arguments, that's not common naming style for setLogger

youssefsoli replied:
it causes a naming conflict with the logging macro, since the macro calls `logIdentifier()`, the local variable is used instead which doesn't build.

jyavenard commented on Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
```diff
@@ -536,7 +531,10 @@ WebMParser::WebMParser(Callback& callback)
-WebMParser::~WebMParser() = default;
+WebMParser::~WebMParser()
+{
+    ALWAYS_LOG_IF_POSSIBLE(LOGIDENTIFIER);
```
if we log the destructor, should log the constructor

youssefsoli replied:
ditto.

jyavenard commented on Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
```diff
@@ -607,14 +606,16 @@ ExceptionOr<int> WebMParser::parse(SourceBufferParser::Segment&& segment)
 }
 
-void WebMParser::setLogger(const Logger& logger, const void* logIdentifier)
+void WebMParser::setLogger(const Logger& newLogger, const void* newLogIdentifier)
```
don't change the name of argument

youssefsoli replied:
ditto.

youssefsoli committed:
[Cocoa][MSE] Fix SourceBufferParserWebM logging
https://bugs.webkit.org/show_bug.cgi?id=242720

Reviewed by Jean-Yves Avenard.

Currently, the logger for SourceBufferParsers are not being set within
MediaSourcePrivateAVFObjC. Futhermore, the setLogger function within
SourceBufferParserWebM never sets its own logger and log identifier, leading
to no logs ever being outputted.

* Source/WebCore/platform/audio/cocoa/AudioFileReaderCocoa.cpp:
(WebCore::AudioFileReader::demuxWebMData const):
* Source/WebCore/platform/graphics/avfoundation/objc/MediaSourcePrivateAVFObjC.mm:
(WebCore::MediaSourcePrivateAVFObjC::addSourceBuffer):
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.h:
* Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferParserAVFObjC.mm:
(WebCore::SourceBufferParserAVFObjC::~SourceBufferParserAVFObjC):
(WebCore::SourceBufferParserAVFObjC::appendData):
(WebCore::SourceBufferParserAVFObjC::flushPendingMediaData):
(WebCore::SourceBufferParserAVFObjC::setShouldProvideMediaDataForTrackID):
(WebCore::SourceBufferParserAVFObjC::shouldProvideMediadataForTrackID):
(WebCore::SourceBufferParserAVFObjC::resetParserState):
(WebCore::SourceBufferParserAVFObjC::invalidate):
(WebCore::SourceBufferParserAVFObjC::setLogger):
(WebCore::SourceBufferParserAVFObjC::didParseStreamDataAsAsset):
(WebCore::SourceBufferParserAVFObjC::didFailToParseStreamDataWithError):
(WebCore::SourceBufferParserAVFObjC::didProvideMediaDataForTrackID):
(WebCore::SourceBufferParserAVFObjC::willProvideContentKeyRequestInitializationDataForTrackID):
(WebCore::SourceBufferParserAVFObjC::didProvideContentKeyRequestInitializationDataForTrackID):
(WebCore::SourceBufferParserAVFObjC::didProvideContentKeyRequestSpecifierForTrackID):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::WebMParser::~WebMParser):
(WebCore::WebMParser::reset):
(WebCore::WebMParser::setLogger):
(WebCore::WebMParser::logChannel const):
(WebCore::WebMParser::invalidate):
(WebCore::SourceBufferParserWebM::~SourceBufferParserWebM):
(WebCore::SourceBufferParserWebM::setLogger):
(WebCore::SourceBufferParserWebM::setMinimumAudioSampleDuration):
(WebCore::SourceBufferParserWebM::logChannel const):
(WebCore::logChannel): Deleted.
(WebCore::logClassName): Deleted.
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.h:
(WebCore::WebMParser::TrackData::logChannel const):
(WebCore::WebMParser::VideoTrackData::logClassName const):
(WebCore::WebMParser::AudioTrackData::logClassName const):
(WebCore::WebMParser::VideoTrackData::logChannel const): Deleted.
(WebCore::WebMParser::AudioTrackData::logChannel const): Deleted.
(WebCore::WebMParser::loggerPtr const):
(WebCore::SourceBufferParserWebM::loggerPtr const):
(WebCore::WebMParser::logIdentifier const): Deleted.
(WebCore::SourceBufferParserWebM::logIdentifier const): Deleted.
* Source/WebKit/Shared/mac/MediaFormatReader/MediaFormatReader.cpp:
(WebKit::MediaFormatReader::nextLogIdentifier):
(WebKit::MediaFormatReader::logChannel):
(WebKit::nextLogIdentifier): Deleted.
(WebKit::logChannel): Deleted.
(WebKit::logClassName): Deleted.
* Source/WebKit/Shared/mac/MediaFormatReader/MediaFormatReader.h:

Canonical link: https://commits.webkit.org/252482@main
